taskboot/docker.py

```diff
@@ -245,3 +256,37 @@ def _find_replacement(original):
     parser.content = open(dockerfile).read()
     logger.info('Initial parent images: {}'.format(' & '.join(parser.parent_images)))
     parser.parent_images = list(map(_find_replacement, parser.parent_images))
+
+
+def read_manifest(path):
+    '''
+    Read a Docker archive manifest and load it as JSON
+    '''
+    assert os.path.exists(path), 'Missing archive {}'.format(path)
+    assert tarfile.is_tarfile(path), 'Not a TAR archive {}'.format(path)
+
+    tar = tarfile.open(path)
+    manifest_raw = tar.extractfile('manifest.json')
+    return json.loads(manifest_raw.read().decode('utf-8'))
```

Comment:
Curiosity more than anything else, can't json.load be used here with the file object returned by extractfile?